specs/notation-config.md

```diff
@@ -0,0 +1,59 @@
+# Notation Configuration
+
+To enable persisted configuration, simplifying the execution of the `notation` cli, the following configuration file will be available
+
+> Note: there will be a policy based configuration that will come at a later point.
+
+## Location
+
+The default location and file will be stored at: `~/.notary/notation-config.json`. The `notation` cli and libraries will support alternate locations through a `config-location` parameters.
+
+## Properties
+
+- `enabled` - bool, which acts as an on/off switch for default notation behavior. --enabled false would disable all automatic validations
+- `verificationCerts` - collection of name/value pairs for a collection of public certs that are used for verification. These may be replaced with a future policy configuration.
+  - `name` - a named reference to the certificate
+  - `path` - a location by which the certificate can be found by the notation cli or notation libraries
+- `signing-keys` - a collection of name/value pairs of signing keys.
+  - `name` - a named reference to the key
+  - `path` - a location by which the key can be found by the notation cli or notation libraries
+  - `default` - the signing key to be used when `notation sign` is called without `--name`
+- `insecureRegistries` - a list of registries that may be used without https
+
+## Samples 
+
+`~/.notary/notation-config.json`
+
+```json
+{
+  "enabled": true,
+  "verificationCerts": {
+    "certs": [
+      {
+        "name": "wabbit-networks.io",
+        "path": "~/./notary/keys/wabbit-networks.crt"
+      },
+      {
+        "name": "import.acme-rockets.io",
+        "path": "~/./notary/keys/import-acme-rockets.crt"
+      }
+    ]
+  },
```

Comment:
Why do we need `certs` inside of `verificationCerts`? It is redundant.

Reply:
I was trying to align with `keys` which has `default` nested, so it needed `keys`. 
Suggestions?

Reply:
Leaving for this PR, and lets address how we want to handle `verificationCerts` and `signingKeys`
Do we want to pull default out to a root node?

Reply:
@SteveLasker - Moving this into the notary location exposes the notation CLI to need ACLs and policise that are present with notaryV1. My recommendatiaon is to have this under .notation for now and open an issue to unify the config location with notaryv1 if that is the requirement. 
/cc @gokarnm

Reply:
> is a requirement, and the motivation for unifying config with notary V1.

It was more about reducing folder sprawl. I'm trying to avoid end users having to "feel the pain" of multiple folders being created as a result of using v1 and v2. 
We don't have much usage of notary v1, so it's a minimal issue. 
It's not the most pressing issue, so you could also argue, if most users don't have notary v1, is it really folder sprawl as they'll only have a single folder, "./notation".
I'll update to use notation and not be something to take more time for us to think about

Reply:
> are we thinking of supporting truststore and trustpolicy evaluation

Yes, we should incorporate truststore policy. However, suggesting we add that as we iterate the details of the trust store policy. This will document the current state, so we can release a notation cli that we can iterate with. 
As we complete the truststore policy design and implementation, we can update this config spec.